Restaurant Cost Food 

Project Description

Who doesn’t love food? All of us must have craving for at least a few favourite food items, we may also have a few places where we like to get them, a restaurant which serves our favourite food the way we want it to be. But there is one factor that will make us reconsider having our favourite food from our favourite restaurant, the cost. Here in this hackathon, you will be predicting the cost of the food served by the restaurants across different cities in India. You will use your Data Science skills to investigate the factors that really affect the cost, and who knows maybe you will even gain some very interesting insights that might help you choose what to eat and from where.

we are using with following 2 files
1.     train.csv : Use this dataset to train the model. This file contains all the details related to restaurant food cost as well as the target variable “cost”. You have to train your model using this file.
2.     test.csv : Use the trained model to predict the cost of a two person meal.


Dataset Attributes
TITLE: The feature of the restaurant which can help identify what and for whom it is suitable for.
RESTAURANT_ID: A unique ID for each restaurant.
CUISINES: The variety of cuisines that the restaurant offers.
TIME: The open hours of the restaurant.
CITY: The city in which the restaurant is located.
LOCALITY: The locality of the restaurant.
RATING: The average rating of the restaurant by customers.
VOTES: The overall votes received by the restaurant.
COST: The average cost of a two-person meal.


In [6]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [4]:
df_test= pd.read_excel(r'C:\Users\USER\Downloads\Data_Test.xlsx')

In [5]:
df_test.head()

TITLE  RESTAURANT_ID                                      CUISINES  \
0  CASUAL DINING           4085         North Indian, Chinese, Mughlai, Kebab   
1    QUICK BITES          12680  South Indian, Fast Food, Pizza, North Indian   
2  CASUAL DINING           1411       North Indian, Seafood, Biryani, Chinese   
3           None            204                                       Biryani   
4    QUICK BITES          13453                          South Indian, Kerala   

                                                TIME       CITY      LOCALITY  \
0                      12noon – 12midnight (Mon-Sun)      Noida     Sector 18   
1                            7am – 12:30AM (Mon-Sun)     Mumbai    Grant Road   
2                           11am – 11:30pm (Mon-Sun)     Mumbai  Marine Lines   
3  9am – 10pm (Mon, Wed, Thu, Fri, Sat, Sun), 10:...  Faridabad           NIT   
4                              11am – 10pm (Mon-Sun)      Kochi        Kaloor   

  RATING       VOTES  
0    4.3   564 votes  
1    4.2    61 votes  
2    3.8   350 votes  
3    3.8  1445 votes  
4    3.6    23 votes

In [10]:
df_test.shape

(4231, 8)

In [12]:
df_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4231 entries, 0 to 4230
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   TITLE          4231 non-null   object
 1   RESTAURANT_ID  4231 non-null   int64 
 2   CUISINES       4231 non-null   object
 3   TIME           4231 non-null   object
 4   CITY           4196 non-null   object
 5   LOCALITY       4201 non-null   object
 6   RATING         4229 non-null   object
 7   VOTES          3829 non-null   object
dtypes: int64(1), object(7)
memory usage: 264.6+ KB


In [7]:
df_train= pd.read_excel(r'C:\Users\USER\Downloads\Data_Train.xlsx')

In [9]:
df_train.head()

TITLE  RESTAURANT_ID  \
0      CASUAL DINING           9438   
1  CASUAL DINING,BAR          13198   
2      CASUAL DINING          10915   
3        QUICK BITES           6346   
4     DESSERT PARLOR          15387   

                                     CUISINES  \
0                 Malwani, Goan, North Indian   
1              Asian, Modern Indian, Japanese   
2  North Indian, Chinese, Biryani, Hyderabadi   
3                            Tibetan, Chinese   
4                                    Desserts   

                                     TIME     CITY        LOCALITY RATING  \
0  11am – 4pm, 7:30pm – 11:30pm (Mon-Sun)    Thane  Dombivali East    3.6   
1                    6pm – 11pm (Mon-Sun)  Chennai       Ramapuram    4.2   
2     11am – 3:30pm, 7pm – 11pm (Mon-Sun)  Chennai      Saligramam    3.8   
3                 11:30am – 1am (Mon-Sun)   Mumbai     Bandra West    4.1   
4                    11am – 1am (Mon-Sun)   Mumbai     Lower Parel    3.8   

       VOTES  COST  
0   49 votes  1200  
1   30 votes  1500  
2  221 votes   800  
3   24 votes   800  
4  165 votes   300

In [13]:
df_train.shape

(12690, 9)

In [15]:
df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12690 entries, 0 to 12689
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   TITLE          12690 non-null  object
 1   RESTAURANT_ID  12690 non-null  int64 
 2   CUISINES       12690 non-null  object
 3   TIME           12690 non-null  object
 4   CITY           12578 non-null  object
 5   LOCALITY       12592 non-null  object
 6   RATING         12688 non-null  object
 7   VOTES          11486 non-null  object
 8   COST           12690 non-null  int64 
dtypes: int64(2), object(7)
memory usage: 892.4+ KB


Observation:
Train data shape: 12690 rows, 9 columns
Test data shape: 4231 rows, 8 columns



In [16]:
df_train['COST'].value_counts()

300      1595
400      1357
500      1300
200      1031
600       806
         ... 
8000        1
14000       1
5200        1
80          1
70          1
Name: COST, Length: 86, dtype: int64

In [17]:
df_test['VOTES'].value_counts()

6 votes       30
22 votes      26
58 votes      26
9 votes       24
42 votes      24
              ..
1206 votes     1
796 votes      1
1720 votes     1
734 votes      1
1239 votes     1
Name: VOTES, Length: 1136, dtype: int64

In [18]:
df_train_without_label = df_train.drop(columns=["COST"] , inplace=False)

In [19]:
df_train_without_label.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12690 entries, 0 to 12689
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   TITLE          12690 non-null  object
 1   RESTAURANT_ID  12690 non-null  int64 
 2   CUISINES       12690 non-null  object
 3   TIME           12690 non-null  object
 4   CITY           12578 non-null  object
 5   LOCALITY       12592 non-null  object
 6   RATING         12688 non-null  object
 7   VOTES          11486 non-null  object
dtypes: int64(1), object(7)
memory usage: 793.3+ KB


In [20]:
df_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4231 entries, 0 to 4230
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   TITLE          4231 non-null   object
 1   RESTAURANT_ID  4231 non-null   int64 
 2   CUISINES       4231 non-null   object
 3   TIME           4231 non-null   object
 4   CITY           4196 non-null   object
 5   LOCALITY       4201 non-null   object
 6   RATING         4229 non-null   object
 7   VOTES          3829 non-null   object
dtypes: int64(1), object(7)
memory usage: 264.6+ KB


In [21]:
df = pd.concat([df_train_without_label,df_test],axis=0)
df.tail()

TITLE  RESTAURANT_ID  \
4226  CASUAL DINING           9057   
4227           None           1247   
4228    QUICK BITES           8617   
4229    QUICK BITES           6485   
4230    QUICK BITES           2421   

                                           CUISINES  \
4226                 North Indian, Mughlai, Chinese   
4227  Biryani, North Indian, Sandwich, Salad, Wraps   
4228                      Continental, North Indian   
4229                               Rolls, Beverages   
4230                                        Mughlai   

                                                   TIME         CITY  \
4226                        11:30am – 11:30pm (Mon-Sun)    New Delhi   
4227                               11am – 1am (Mon-Sun)    Bangalore   
4228                         9:30am – 10:30pm (Mon-Sun)    Faridabad   
4229  11am – 11:30pm (Mon, Tue, Wed, Thu, Sat, Sun),...        Kochi   
4230                                5pm – 1am (Mon-Sun)  Bandra West   

              LOCALITY RATING      VOTES  
4226      Punjabi Bagh    3.9  287 votes  
4227        HSR Layout    4.3  469 votes  
4228         Sector 86    3.7   53 votes  
4229             Kochi      -        NaN  
4230  Perry Cross Road    3.7   63 votes

In [22]:
df.shape

(16921, 8)

In [23]:
df.nunique()

TITLE              123
RESTAURANT_ID    15574
CUISINES          5183
TIME              3295
CITY               450
LOCALITY          1636
RATING              32
VOTES             2075
dtype: int64

In [24]:
df.drop_duplicates()

TITLE  RESTAURANT_ID  \
0         CASUAL DINING           9438   
1     CASUAL DINING,BAR          13198   
2         CASUAL DINING          10915   
3           QUICK BITES           6346   
4        DESSERT PARLOR          15387   
...                 ...            ...   
4226      CASUAL DINING           9057   
4227               None           1247   
4228        QUICK BITES           8617   
4229        QUICK BITES           6485   
4230        QUICK BITES           2421   

                                           CUISINES  \
0                       Malwani, Goan, North Indian   
1                    Asian, Modern Indian, Japanese   
2        North Indian, Chinese, Biryani, Hyderabadi   
3                                  Tibetan, Chinese   
4                                          Desserts   
...                                             ...   
4226                 North Indian, Mughlai, Chinese   
4227  Biryani, North Indian, Sandwich, Salad, Wraps   
4228                      Continental, North Indian   
4229                               Rolls, Beverages   
4230                                        Mughlai   

                                                   TIME         CITY  \
0                11am – 4pm, 7:30pm – 11:30pm (Mon-Sun)        Thane   
1                                  6pm – 11pm (Mon-Sun)      Chennai   
2                   11am – 3:30pm, 7pm – 11pm (Mon-Sun)      Chennai   
3                               11:30am – 1am (Mon-Sun)       Mumbai   
4                                  11am – 1am (Mon-Sun)       Mumbai   
...                                                 ...          ...   
4226                        11:30am – 11:30pm (Mon-Sun)    New Delhi   
4227                               11am – 1am (Mon-Sun)    Bangalore   
4228                         9:30am – 10:30pm (Mon-Sun)    Faridabad   
4229  11am – 11:30pm (Mon, Tue, Wed, Thu, Sat, Sun),...        Kochi   
4230                                5pm – 1am (Mon-Sun)  Bandra West   

              LOCALITY RATING      VOTES  
0       Dombivali East    3.6   49 votes  
1            Ramapuram    4.2   30 votes  
2           Saligramam    3.8  221 votes  
3          Bandra West    4.1   24 votes  
4          Lower Parel    3.8  165 votes  
...                ...    ...        ...  
4226      Punjabi Bagh    3.9  287 votes  
4227        HSR Layout    4.3  469 votes  
4228         Sector 86    3.7   53 votes  
4229             Kochi      -        NaN  
4230  Perry Cross Road    3.7   63 votes  

[16879 rows x 8 columns]